### PR TITLE
Ad 64 Step 2 additional targeting select

### PR DESCRIPTION
**Pull request by KonradFak**

### CODE REVIEW

File: src/app/common/components/targeting-select/targeting-select.component.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+  ngDoCheck() {
+    const addedItemsChanged = this.arrayDiffer.diff(this.addedItems);
+
+    if (addedItemsChanged) {
+      this.selectedItems = this.selectedItems.filter((item) => item.selected);
+    }
+  }
+
+  changetViewModel(options) {
```
Comment (BartoszBilejczyk) [inline src/app/common/components/targeting-select/targeting-select.component.ts]: possible typo, unnecessary 't' after change - search for it everywhere

**Comment by aqum** [comment]
Comment from our offline discussion:
- try to not relay on object or array reference - if there will be always new reference you probably won't need any differ strategies
- try to move logic for adding/removing elements to parent component from list using `@Output` to keep it more reusable/testable and `dumb`

File: src/app/models/targeting-option.model.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+interface targetingOptionModel {
```
Comment (aqum) [inline src/app/models/targeting-option.model.ts]: There is convention to name interfaces from large letter

File: src/app/advertiser/edit-campaign/edit-campaign-additional-targeting/edit-campaign-additional-targeting.component.ts [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
   constructor(private route: ActivatedRoute) {
+    this.TargetingOptionsToAdd = cloneDeep(this.route.snapshot.data.targetingOptions.criteria);
+    this.TargetingOptionsToExclude = cloneDeep(this.route.snapshot.data.targetingOptions.criteria);
+
     this.route.queryParams.subscribe(params => this.goesToSummary = params.summary);
-  };
+  }
+
+  updateAddedItems(items) {
+    this.addedItems = [];
```
Comment (aqum) [inline src/app/advertiser/edit-campaign/edit-campaign-additional-targeting/edit-campaign-additional-targeting.component.ts]: I think cleaner would be to write `this.addedItems = [...items]`

**Pull request merged**